server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -407,27 +377,36 @@ public String executor() {
                 return ThreadPool.Names.GENERIC;
             }
         };
-        try {
-            logger.info("Sending extension request type: " + REQUEST_EXTENSION_ACTION_NAME);
-            transportService.connectToExtensionNode(extension);
-            transportService.sendRequest(
-                extension,
-                REQUEST_EXTENSION_ACTION_NAME,
-                new InitializeExtensionRequest(transportService.getLocalNode(), extension),
-                initializeExtensionResponseHandler
-            );
-            inProgressFuture.orTimeout(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS).join();
-        } catch (CompletionException | ConnectTransportException e) {
-            if (e.getCause() instanceof TimeoutException || e instanceof ConnectTransportException) {
-                logger.info("No response from extension to request.", e);
-            } else if (e.getCause() instanceof RuntimeException) {
-                throw (RuntimeException) e.getCause();
-            } else if (e.getCause() instanceof Error) {
-                throw (Error) e.getCause();
-            } else {
-                throw new RuntimeException(e.getCause());
+
+        logger.info("Sending extension request type: " + REQUEST_EXTENSION_ACTION_NAME);
+        transportService.getThreadPool().generic().execute(new AbstractRunnable() {
+            @Override
+            public void onFailure(Exception e) {
+                extensionIdMap.remove(extension.getId());
+                if (e.getCause() instanceof ConnectTransportException) {
+                    logger.info("No response from extension to request.", e);
+                    throw (ConnectTransportException) e.getCause();
```

Comment:
We should not be throwing any exceptions in `onFailure`.  We're already in a catch block for `run()` when handling these.

Reply:
These exceptions are required. If a connection to an extension is not successful, the flow will come down to `onFailure` to return:
`java.lang.RuntimeException: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:4500
`

Reply:
OK.   In which case we should have just overridden `run()` instead of doRun() since that's the method that will eventually throw this exception.   In any case, we can fine tune this later, I think you were going to submit an issue proposing a way to do this with transport actions, so this working code is fine for now.

Reply:
https://github.com/opensearch-project/OpenSearch/issues/8204